homeassistant/components/mystrom/switch.py

```diff
@@ -28,34 +31,55 @@
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the myStrom entities."""
+    info = hass.data[DOMAIN][entry.entry_id].info
+    device_type = info["type"]
+
+    if device_type in [101, 106, 107]:
+        device = hass.data[DOMAIN][entry.entry_id].device
+        async_add_entities([MyStromSwitch(device, entry.title)])
+
+
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the myStrom switch/plug integration."""
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-
-    try:
-        plug = _MyStromSwitch(host)
-        await plug.get_state()
-    except MyStromConnectionError as err:
-        _LOGGER.error("No route to myStrom plug: %s", host)
-        raise PlatformNotReady() from err
-
-    async_add_entities([MyStromSwitch(plug, name)])
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.12.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+    )
 
 
 class MyStromSwitch(SwitchEntity):
     """Representation of a myStrom switch/plug."""
 
-    def __init__(self, plug, name):
+    def __init__(self, plug, name) -> None:
```

Comment:
Please avoid partial type hints.
This one (and the same in light) could be done in a separate (parallel) PR.

Reply:
removed type hints (for this PR)